src/Native/Unix/System.Native/pal_process.c

```diff
@@ -221,6 +232,34 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
         goto done;
     }
 
+#if HAVE_VFORK_SHM
+    // This platform has the shared-memory implementation of vfork(). For a one gigabyte process
+    // the expected performance gain of using vfork() rather than fork() is 99.5% merely due to
+    // avoiding page faults as the kernel does not need to set all writable pages in the parent
+    // process to copy-on-write because the child process is allowed to write to the parent process
+    // memory pages. Eliding the pipe for sending the error from execve() back to the parent is pure
+    // gravy.
+
+    // The thing to remember about shared memory vfork() is the documentation is way out of date.
+    // It does the following things:
+    // * creates a new process in the memory space of the calling process.
+    // * blocks the calling thread (not process!) in an uninterruptable sleep
+    // * sets up the process records so the following happen:
+    //   + execve() replaces the memory space in the child and unblocks the parent
+    //   + process exit by any means unblocks the parent
+    //   + ptrace() makes a security demand against the parent process
+    //   + accessing the terminal with read() or write() fail in system-dependent ways
+    // Due to lack of documentation, changing signals in the vfork() child is a bad idea. We don't
+    // do this, but it's worth pointing out.
```

Comment:
Can you rewrite this and the next paragraphs as a list of things we mustn't do after the vfork?

> + ptrace() makes a security demand against the parent process

I don't know what it means and why it is important to mention.